syntax-test_dev.py

```diff
@@ -0,0 +1,271 @@
+    assertion_colrange = None
+    if starts_with_comment_token:
+        assertion = re.match(r'\s*(?:(<-)|(\^+))', line_text[starts_with_comment_token.end():])
+        if assertion:
+            if assertion.group(1):
+                assertion_colrange = (starts_with_comment_token.start(1),
+                                      starts_with_comment_token.start(1))
+            elif assertion.group(2):
+                assertion_colrange = (starts_with_comment_token.end() + assertion.start(2),
+                                      starts_with_comment_token.end() + assertion.end(2))
+
+    return (starts_with_comment_token, assertion_colrange, line_region)
+
+
+def is_syntax_test_line(view, pos, must_contain_assertion):
+    """Determine whether the line at the given character position is a syntax test line.
+    It can optionally treat lines with comment markers but no assertion as a syntax test,
+    useful for while the line is being written.
+    """
+
+    starts_with_comment_token, assertion_cols, _ = get_details_of_test_assertion_line(view, pos)
+    if starts_with_comment_token:
+        return not must_contain_assertion or assertion_cols is not None
+    return False
+
+
+def get_details_of_line_being_tested(view):
+    """Given a view, work from the cursor upwards to find all syntax test lines that occur
+    before the line that is being tested.
+    Return a tuple containing a list of assertion line details,
+    along with the region of the line being tested."""
+
+    if not is_syntax_test_file(view):
+        return (None, None)
+
+    lines = []
+    pos = view.sel()[0].begin()
+    first_line = True
+    while pos >= 0:
+        line = get_details_of_test_assertion_line(view, pos)
+        pos = line[2].begin() - 1
+        if line[1]:
+            lines.append(line)
+        elif not first_line or not line[0]:
+            break
+        elif line[0]:
+            lines.append(line)
+        first_line = False
+
+    if len(lines) == 0:
+        return (None, line[2])
+    else:
+        return (lines, line[2])
+
+
+class SyntaxTestEventListener(sublime_plugin.EventListener):
+    """Event Listener for responding to keybinding contexts"""
+
+    def on_query_context(self, view, key, operator, operand, match_all):
+        # all contexts supported will have boolean results, so ignore regex operators
+        if operator not in (sublime.OP_EQUAL, sublime.OP_NOT_EQUAL):
+            return None
+
+        def current_line_is_a_syntax_test():
+            return is_syntax_test_line(view, view.sel()[0].begin(), False)
+
+        def line_above_is_a_syntax_test():
+            line = get_details_of_test_assertion_line(view, view.sel()[0].begin())
+            if line[0] is None:
+                return False
+            else:
+                return is_syntax_test_line(view, line[2].begin() - 1, True)
+
+        def file_contains_syntax_tests():
+            return is_syntax_test_file(view)
+
+        if key not in locals():
```

Comment:
thanks for the info - I don't seem to get notifications about comments on my gists.